torch/__init__.py

```diff
@@ -26,39 +27,49 @@
 # Load the extension module
 ################################################################################
 
-# Loading the extension with RTLD_GLOBAL option allows to not link extension
-# modules against the _C shared object. Their missing THP symbols will be
-# automatically filled by the dynamic loader.
-import os as _dl_flags
-
 # if we have numpy, it *must* be imported before the call to setdlopenflags()
 # or there is risk that later c modules will segfault when importing numpy
 try:
     import numpy as np
 except:
     pass
 
-# first check if the os package has the required flags
-if not hasattr(_dl_flags, 'RTLD_GLOBAL') or not hasattr(_dl_flags, 'RTLD_NOW'):
-    try:
-        # next try if DLFCN exists
-        import DLFCN as _dl_flags
-    except ImportError:
-        # as a last attempt, use compile-time constants
-        import torch._dl as _dl_flags
+if platform.system() == 'Windows':
+    import os
+    os.environ['PATH'] = os.path.dirname(__file__) + '\\lib\\;' + os.environ['PATH']
+
+    from torch._C import *
+    __all__ += [name for name in dir(_C)
+                if name[0] != '_' and
+                not name.endswith('Base')]
```

Comment:
This code seems to be the same in both cases, so why replicate it (I mean `__all__` not the import)?